### ISSUE TITLE
Core dumped with pytorch 0.2.0 and multi-gpu

### ISSUE THREAD

**Issue by oldgil** [junyanz/pytorch-CycleGAN-and-pix2pix#114]
Does anyone has the same problem? It often occurs randomly. By the way, this code works well in previous pytorch versions, but it will take up more memory.

**Comment by junyanz** [comment]
I haven't tested the multi-gpu with the latest version. Could you give more details on your errors?

**Comment by LambdaWill** [comment]
I confirm this.
```bash
  File "train.py", line 48, in <module>
    model.optimize_parameters() # update parameters
  File "/home/clp001/pt-lr/models/pix2pix_model.py", line 229, in optimize_parameters
    self.backward_D()
  File "/home/clp001/pt-lr/models/pix2pix_model.py", line 189, in backward_D
    self.loss_D.backward()
  File "/home/clp001/anaconda3/lib/python3.6/site-packages/torch/autograd/variable.py", line 156, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, retain_variables)
  File "/home/clp001/anaconda3/lib/python3.6/site-packages/torch/autograd/__init__.py", line 98, in backward
    variables, grad_variables, retain_graph)
  File "/home/clp001/anaconda3/lib/python3.6/site-packages/torch/nn/parallel/_functions.py", line 25, in backward
    return comm.reduce_add_coalesced(grad_outputs, self.input_device)
  File "/home/clp001/anaconda3/lib/python3.6/site-packages/torch/cuda/comm.py", line 122, in reduce_add_coalesced
    result = reduce_add(flattened, destination)
  File "/home/clp001/anaconda3/lib/python3.6/site-packages/torch/cuda/comm.py", line 92, in reduce_add
    nccl.reduce(inputs, outputs, root=destination)
  File "/home/clp001/anaconda3/lib/python3.6/site-packages/torch/cuda/nccl.py", line 161, in reduce
    assert(root >= 0 and root < len(inputs))
AssertionError
```

**Comment by SsnL** [comment]
@oldgil  @LambdaWill  IIRC, there was a related DataParallel bug that is fixed between 0.2 and 0.3. Could you try to upgrade pytorch to 0.3?
